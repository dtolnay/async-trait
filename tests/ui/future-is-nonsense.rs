use async_trait::async_trait;

#[async_trait]
trait Trait {
    #[future_is[Nonsense]]
    async fn method1(&mut self);

    #[future_is[Sync - Send]]
    async fn method2(&mut self);
    
    #[future_is(Sync)]
    async fn method3(&mut self);
}

fn main() {}
