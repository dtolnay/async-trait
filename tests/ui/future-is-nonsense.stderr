error: expected `+`
 --> $DIR/future-is-nonsense.rs:8:22
  |
8 |     #[future_is[Sync - Send]]
  |                      ^

error: expected square brackets
  --> $DIR/future-is-nonsense.rs:11:16
   |
11 |     #[future_is(Sync)]
   |                ^

error[E0405]: cannot find trait `Nonsense` in this scope
 --> $DIR/future-is-nonsense.rs:5:17
  |
4 | trait Trait {
  |            - help: you might be missing a type parameter: `<Nonsense>`
5 |     #[future_is[Nonsense]]
  |                 ^^^^^^^^ not found in this scope
